huggingface_sb3/push_to_hub.py

```diff
@@ -7,30 +7,309 @@
+    :param repo_local_path: path of the local repository
+    """
+    # Step 1: Evaluate the agent
+    mean_reward, std_reward = evaluate_policy(model, eval_env, n_eval_episodes, is_deterministic)
+
+    # Step 2: Create json evaluation
+    evaluate_data = {
+        "mean_reward": mean_reward,
+        "std_reward": std_reward,
+        "is_deterministic": is_deterministic,
+        "n_eval_episodes": n_eval_episodes
+    }
+
+    # Step 3: Write a JSON file
+    with open(Path(repo_local_path) / 'results.json', 'w') as outfile:
+        json.dump(evaluate_data, outfile)
+
+    print("results", mean_reward)
+    return mean_reward, std_reward
+
+
+def is_atari(env_id: str) -> bool:
+    """
+    Check if the environment is an Atari one
+    (Taken from RL-Baselines3-zoo)
+    :param env_id: name of the environment
+    """
+    entry_point = gym.envs.registry.env_specs[env_id].entry_point
+    return "AtariEnv" in str(entry_point)
+
+
+def _generate_replay(model, eval_env, video_length, is_deterministic, repo_local_path):
+    """
+    Generate a replay video of the agent
+    :param model: name of the model
+    :param eval_env: environment used to evaluate the agent
+    :param video_length: length of the video (in timesteps)
+    :param is_deterministic: use deterministic or stochastic actions
+    :param repo_local_path: path of the local repository
+    """
+    # Step 1: Create the VecVideoRecorder
+    env = VecVideoRecorder(
+        eval_env,
+        "./",  # Temporary video folder
+        record_video_trigger=lambda x: x == 0,
+        video_length=video_length,
+        name_prefix="",
+    )
+
+    obs = env.reset()
+    env.reset()
+    try:
+        for _ in range(video_length + 1):
+            action, _ = model.predict(obs, deterministic=is_deterministic)
+            obs, _, _, _ = env.step(action)
+            # env.render()
+
+        # Save the video
+        env.close()
+
+        # Rename the video
+        os.rename(env.video_recorder.path, "test.mp4")
+
+        # Convert the video with x264 codec
+        inp = "./test.mp4"
+        out = "replay.mp4"
+        os.system(f"ffmpeg -y -i {inp} -vcodec h264 {out}".format(inp, out))
+        """
+        call([ffmpeg,
+              '-i', "./replay.mp4",
+              '-vcodec', h264,
+              replay.mp4])
+        """
+        # Move the video
+        shutil.move(os.path.join("./", "replay.mp4"), os.path.join(repo_local_path, "replay.mp4"))
+    except KeyboardInterrupt:
+        pass
+    except:
+        # Add a message for video
+        print("We are unable to generate a replay of your agent")
+        print("Please send a message to thomas.simonini@huggingface.co")
+
+
+def select_tags(env_id):
+    """
+    Define the tags for the model card
+    :param env_id: name of the environment
+    """
+
+    model_card = f"""
+---
+      tags:
+      - {env_id}
+      - deep-reinforcement-learning
+      - reinforcement-learning
+      - stable-baselines3
+---
```

Comment:
Why not use `README_TEMPLATE` instead for this?

Reply:
Because `{env_id}` is not in `README_TEMPLATE`